custodia/store/sqlite.py

```diff
@@ -110,8 +110,11 @@ def list(self, keyfilter=''):
                     continue
                 if not row[0].startswith(child_prefix):
                     continue
-                value.append(row[0][len(child_prefix):].lstrip('/'))
-
+                result = row[0][len(child_prefix):].lstrip('/')
+                if not self.get(row[0]):
```

Comment:
This is incorrect you already have (or can easily get) the value in row, why get it again ?
Check if the search returns the value and if not make it return the value too, then just check the value directly.